docs/content/querying/lookups.md

```diff
@@ -296,3 +296,262 @@ To test this setup, you can send key/value pairs to a kafka stream via the follo
 ```
 
 Renames can then be published as `OLD_VAL->NEW_VAL` followed by newline (enter or return)
+
+Dynamic configuration (EXPERIMENTAL)
+------------------------------------
+
+The following documents the behavior of the cluster-wide config which is accessible through the coordinator.
+The configuration is propagated through the concept of "tier" of servers.
+A "tier" is defined as a group of services which should receive a set of lookups.
+For example, you might have all historicals be part of `__default`, and Peons be part of individual tiers for the datasources they are tasked with.
+The tiers for lookups are completely independent of historical tiers.
+
+These configs are accessed using JSON through the following URI template
+
+```
+http://<COORDINATOR_IP>:<PORT>/druid/coordinator/v1/lookups/{tier}/{id}
+```
+
+All URIs below are assumed to have `http://<COORDINATOR_IP>:<PORT>` prepended.
+
+If you have NEVER configured lookups before, you MUST post an empty json object `{}` to `/druid/coordinator/v1/lookups` to initialize the configuration.
+
+These endpoints will return one of the following results:
+
+* 404 if the resource is not found
+* 400 if there is a problem in the formatting of the request
+* 202 if the request was accepted asynchronously (`POST` and `DELETE`)
+* 200 if the request succeeded (`GET` only)
+
+## Configuration propagation behavior
+The configuration is propagated to the historical nodes by the coordinator.
+The query serving nodes (broker / router / peon / historical) have an internal API for managing `POST`/`GET`/`DELETE` of lookups.
+The coordinator periodically checks the dynamic configuration for changes and, when it detects a change it does the following:
+
+1. Post all lookups for a tier to all Druid nodes within that tier.
+2. Delete lookups from a tier which were dropped between the prior configuration values and this one.
+
+If there is no configuration change, the coordinator checks for any nodes which might be new since the last time it propagated lookups and adds all lookups for that node (assuming that node's tier has lookups).
+If there are errors while trying to add or update configuration on a node, that node is temporarily skipped until the next management period. The next management period the update will attempt to be propagated again.
+If there is an error while trying to delete a lookup from a node (or if a node is down when the coordinator is propagating the config), the delete is not attempted again. In such a case it is possible that a node has lookups that are no longer managed by the coordinator.
```

Comment:
That had been discussed as a kind of "dictator" mode, but is not in this MVP pr.

Right now it is in a friendly mode where you can add lookups to a node manually which are not part of the dynamic config.

Adding such a feature as you describe would make it enforce its view of state on all nodes instead of allowing other things to modify the node state. For this PR I went with not enforcing dictator mode.